storage/storage_image.go

```diff
@@ -413,6 +418,12 @@ func (s *storageImageDestination) Reference() types.ImageReference {
 
 // Close cleans up the temporary directory.
 func (s *storageImageDestination) Close() error {
+	for _, v := range s.diffOutputs {
+		if v.Target != "" {
+			os.RemoveAll(v.Target)
```

Comment:
Are these c/image-owned temporary files? If so, it would be simpler to have them in `s.directory`

Are they c/storage-owned? If so, shouldn’t there be an API so that c/storage can manage its own locking and metadata and post-sudden-reboot cleanup? Just removing files managed by another codebase feels very risky.

Are they something else?

Reply:
added a new API to c/storage to deal with the cleanup